inc/views/pluggable/pagination.php

```diff
@@ -143,6 +147,155 @@ public function filter_localization( $data ) {
 		return $data;
 	}
 
+	/**
+	 * Check if Jump to Page feature is enabled.
+	 * 
+	 * @param string $post_type The post type in the loop.
+	 * 
+	 * @return bool 
+	 */
+	private function jump_to_enabled( $post_type = '' ) {
+
+		$allowed = $this->post_types_with_pagination_control();
+		
+		// Use the default blog pagination option if we don't have a specific setting for the archive in question.
+		$mod_id  = $allowed[ $post_type ] ?? 'neve_pagination_type'; 
+		$enabled = get_theme_mod( $mod_id, 'number' ) === 'jump-to-page' ? true : false;
+
+		return apply_filters( 'neve_jump_to_enabled', $enabled, $post_type );
+
+	}
+
+	/**
+	 * An array of Post types that have their own pagination control in Customizer.
+	 * 
+	 * Add post types where you give users the ability to change pagination type for archive pages.
+	 * 
+	 * @return mixed 
+	 */
+	private function post_types_with_pagination_control() {
+
+		$control_ids = array(
+			'post'    => 'neve_pagination_type',
+			'product' => 'neve_shop_pagination_type',
+		);
+
+		return apply_filters( 'neve_pagination_control_post_types', $control_ids );
+	}
+
+	/**
+	 * Filter the pagination links to decide whether or not to show the "Jump to Page" input field.
+	 * 
+	 * @param mixed $markup the pagination links from the filter.
+	 * 
+	 * @return mixed
+	 */
+	public function maybe_add_jump_to_page_input( $markup ) {
+		
+		$post_type = get_post_type();
+
+		if ( $this->jump_to_enabled( $post_type ) ) {
+			return $this->normalize_jump_to_input();    
+		}
+
+		return $markup;
+
+	}
+
+	/**
+	 * Expose our Jump to JavaScript on the page.
+	 * 
+	 * If we don't create a dedicated JS function for the inline onclick, the a tag would first scroll to the top of the page(#) and then submit the form.
+	 * 
+	 * @return void
+	 */
+	public function expose_jump_to_js() {
+
+		echo "
```

Comment:
you should use wp_add_inline_script which is recommended for this kind of actions.